includes/class-reader-activation.php

```diff
@@ -0,0 +1,328 @@
+			$auth_intention = \sanitize_email( $_COOKIE[ self::AUTH_INTENTION_COOKIE ] );
+		}
+		/**
+		 * Filters the session auth intention email address.
+		 *
+		 * @param string|null $auth_intention Email address or null if not set.
+		 */
+		return \apply_filters( 'newspack_auth_intention', $auth_intention );
+	}
+
+	/**
+	 * Whether the user is a reader.
+	 *
+	 * @param \WP_User $user User object.
+	 *
+	 * @return bool Whether the user is a reader.
+	 */
+	public static function is_user_reader( $user ) {
+		$is_reader = (bool) \get_user_meta( $user->ID, self::READER, true );
+
+		if ( false === $is_reader ) {
+			/**
+			 * Filters the roles that can determine if a user is a reader.
+			 *
+			 * @param string[] $roles Array of user roles.
+			 */
+			$reader_roles = \apply_filters( 'newspack_reader_user_roles', [ 'subscriber', 'customer' ] );
+			if ( ! empty( $reader_roles ) ) {
+				$user_data = \get_userdata( $user->ID );
+				$is_reader = ! empty( array_intersect( $reader_roles, $user_data->roles ) );
+			}
+		}
+
+		/**
+		 * Filters whether the user is a reader.
+		 *
+		 * @param bool     $is_reader Whether the user is a reader.
+		 * @param \WP_User $user      User object.
+		 */
+		return \apply_filters( 'newspack_is_user_reader', $is_reader, $user );
+	}
+
+	/**
+	 * Verify email address of a reader given the user.
+	 *
+	 * @param \WP_User $user User object.
+	 *
+	 * @return bool Whether the email address was verified.
+	 */
+	public static function verify_reader_email( $user ) {
+		if ( ! $user ) {
+			return false;
+		}
+
+		/** Should not verify email if user is not a reader. */
+		if ( ! self::is_user_reader( $user ) ) {
+			return false;
+		}
+
+		$verified = \get_user_meta( $user->ID, self::EMAIL_VERIFIED, true );
+		if ( $verified ) {
+			return true;
+		}
+
+		\update_user_meta( $user->ID, self::EMAIL_VERIFIED, true );
+
+		return true;
+	}
+
+	/**
+	 * Check if current reader has its email verified.
+	 *
+	 * @param \WP_User $user User object.
+	 *
+	 * @return bool|null Whether the email address is verified, null if invalid user.
+	 */
+	public static function is_reader_verified( $user ) {
+		if ( ! $user ) {
+			return null;
+		}
+
+		/** Should not verify email if user is not a reader. */
+		if ( ! self::is_user_reader( $user ) ) {
+			return null;
+		}
+
+		return (bool) \get_user_meta( $user->ID, self::EMAIL_VERIFIED, true );
+	}
+
+	/**
+	 * Authenticate a reader session given its user ID.
+	 *
+	 * Warning: this method will only verify if the user is a reader in order to
+	 * authenticate. It will not check for any credentials.
+	 *
+	 * @param int  $user_id      User ID.
+	 * @param bool $verify_email Whether to verify the email address.
+	 *
+	 * @return \WP_User|\WP_Error The authenticated reader or WP_Error if authentication failed.
+	 */
+	private static function set_current_reader( $user_id, $verify_email = false ) {
```

Comment:
This method will later be exposed to be integrated with the magic link functionality (a valid magic link token uses `set_current_reader()` to set the user).

This is meant as a shortcut to set the user and set its email as verified. Magic links are guaranteed to verify the user email.

This is not really necessary now, and maybe not even with magic links because a simple call from the other exposed method can set the verification.

c7fcc8d4173abaec26ba2c5f196feb982c5aac7d removes the option